OpenRA.Game/Graphics/SelectionBarsRenderable.cs

```diff
@@ -80,10 +80,24 @@ Color GetHealthColor(Health health)
 		{
 			if (Game.Settings.Game.TeamHealthColors)
 			{
-				var isAlly = actor.Owner.IsAlliedWith(actor.World.LocalPlayer)
-					|| (actor.EffectiveOwner != null && actor.EffectiveOwner.Disguised
-					&& actor.World.LocalPlayer.IsAlliedWith(actor.EffectiveOwner.Owner));
-				return isAlly ? Color.LimeGreen : actor.Owner.NonCombatant ? Color.Tan : Color.Red;
+				var apparentOwner = actor.EffectiveOwner != null && actor.EffectiveOwner.Disguised
+					? actor.EffectiveOwner.Owner
+					: actor.Owner;
+
+				// for friendly spies, use the normal owner
+				if (actor.Owner.IsAlliedWith(actor.World.LocalPlayer))
```

Comment:
I suspect all these will want to be checking against RenderPlayer to make it behave properly for spectators.